File: public/public-web/src/main/java/ch/difty/scipamato/publ/web/authentication/LogoutPage.java
package ch.difty.scipamato.publ.web.authentication;

import org.apache.wicket.request.mapper.parameter.PageParameters;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.wicketstuff.annotation.mount.MountPath;

import ch.difty.scipamato.common.web.pages.login.AbstractLogoutPage;

@MountPath("logout")
public class LogoutPage extends AbstractLogoutPage<LoginPage> {

    private static final long serialVersionUID = 1L;

    public LogoutPage(@Nullable PageParameters parameters) {
        super(parameters);
    }

    @NotNull
    @Override
    protected LoginPage getResponsePage() {
        return new LoginPage(new PageParameters());
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/authentication/LogoutPage.kt
package ch.difty.scipamato.publ.web.authentication

import ch.difty.scipamato.common.web.pages.login.AbstractLogoutPage
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.wicketstuff.annotation.mount.MountPath

@MountPath("logout")
class LogoutPage(
    parameters: PageParameters?,
) : AbstractLogoutPage<LoginPage>(parameters) {

    override val responsePage: LoginPage
        get() = LoginPage(PageParameters())

    companion object {
        private const val serialVersionUID = 1L
    }
}
